src/cache/cache.rs

```diff
@@ -542,28 +554,41 @@ pub fn storage_from_config(
                     c.bucket, c.endpoint
                 );
 
-                let storage = S3Cache::build(
+                S3Cache::build(
                     &c.bucket,
                     c.region.as_deref(),
                     &c.key_prefix,
                     c.no_credentials,
                     c.endpoint.as_deref(),
                     c.use_ssl,
                 )
-                .map_err(|err| anyhow!("create s3 cache failed: {err:?}"))?;
-
-                return Ok(Arc::new(storage));
+                .map_err(|err| anyhow!("create s3 cache failed: {err:?}"))?
             }
             #[cfg(feature = "webdav")]
             CacheType::Webdav(ref c) => {
                 debug!("Init webdav cache with endpoint {}", c.endpoint);
-                let storage = WebdavCache::build(&c.endpoint, &c.key_prefix)
-                    .map_err(|err| anyhow!("create webdav cache failed: {err:?}"))?;
-                return Ok(Arc::new(storage));
+                WebdavCache::build(&c.endpoint, &c.key_prefix)
+                    .map_err(|err| anyhow!("create webdav cache failed: {err:?}"))?
             }
-            #[allow(unreachable_patterns)]
-            _ => bail!("cache type is not enabled"),
+        };
+
+        let mut op = op.layer(LoggingLayer::default());
+        if !config.disable_cache_retry {
+            // RetryLayer will retry requests when meeting temporary
+            // errors returned by OpenDAL. For example, network
+            // unreachable, timeout and so on.
```

Comment:
is there a list of errors that will trigger a retry ?

Reply:
> is there a list of errors that will trigger a retry ?

They are implementation details that may vary depending on the services.